druid/src/widget/stack.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A widget that arranges its children on top of one another.
+
+use crate::kurbo::{Point, Rect, Size};
+use crate::{
+    BoxConstraints, BoxedWidget, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, UpdateCtx, Widget, WidgetPod,
+};
+
+/// A container that lays out its children along the z-axis, first child at bottom, last child on top.
+#[derive(Default)]
+pub struct Stack<T> {
+    children: Vec<BoxedWidget<T>>,
```

Comment:
I find that we are using this inconsistently, and I'm inclined to remove it; I find the name unclear. I would prefer the more annoying `WidgetPod<T, Box<dyn Widget<T>>` for now, thanks.